### ISSUE TITLE
Device is rejected by the app

### ISSUE THREAD

**Issue by jmevel** [azymohliad/watchmate#32]
Hi !

Today I wanted to flash the newest InfiniTime firmware on my PineTime so I opened WatchMate but it couldn't connect to my existing device.  I get the error message `Device is rejected by the app`.

It seems there's no way delete the device using the app and start again like a fresh new install. I tried to uninstall the app through `Discover` (I'm under KDE), click on `Delete  settings and user data` then reinstall the app but the device is still listed when I reopen the app !

How can I manually delete my device from WatchMate or maybe fix this issue? I'm not yet very familiar with Linux and Flatpak in general, I can't find where WatchMate's config files are located.

Thanks

**Comment by jmevel** [comment]
Thanks a lot for your answer. 

I completely forgot the connection should be made at the device level first before being able to interact with the PineTime through Watchmate. Forgetting the watch from the devices list in my OS and adding it again fixed this issue.

Also I don't know why my watch was still under InfiniTime 1.4.0 (from 2021 !)even though I take care of installing almost every new release. Weird...

I'm closing this issue.
Thanks again for your help (and I appreciate the extra command to collect logs, I'm a Linux beginner and I didn't know such thing)

**Comment by azymohliad** [comment]
Awesome! It's good to know that forgetting the bluetooth device fixed the issue. It would make sense to add a shortcut for that to WatchMate, but it doesn't look like bluer (the bluez library I use) exposes an API for that.

> I completely forgot the connection should be made at the device level first before being able to interact with the PineTime through Watchmate

You mean like pairing it in via the system Bluetooth Manager? That shouldn't be necessary, I think forgetting it from there, and then adding again directly via WatchMate would work the same way.

> Also I don't know why my watch was still under InfiniTime 1.4.0 (from 2021 !)even though I take care of installing almost every new release. Weird...

It looks like you might have missed to validate the firmware from the PineTime settings, see more details [here](https://github.com/InfiniTimeOrg/InfiniTime/blob/main/doc/gettingStarted/updating-software.md#firmware-validation).

**Comment by jmevel** [comment]
> You mean like pairing it in via the system Bluetooth Manager? That shouldn't be necessary, I think forgetting it from there, and then adding again directly via WatchMate would work the same way.

Yes, sorry that's what I mean, I did that using the Bluetooth Manager in the KDE Settings.
I think with earlier versions of WatchMate it wasn't possible to add the device directly through the app. At least I remember it wasn't working for me and the first time I used it, it took me a while to figure out I have to add the device using KDE Bluetooth Manager first (I completely forgot about this when I created this issue ticket).
I haven't tried adding the device directly in WatchMate again, maybe it's actually working now...

> It looks like you might have missed to validate the firmware from the PineTime settings, see more details [here](https://github.com/InfiniTimeOrg/InfiniTime/blob/main/doc/gettingStarted/updating-software.md#firmware-validation).

Thanks, it's exactly what happened, I didn't know the firmware validation feature was automatically resetting the firmware when the device turns off but actually I think I stumbled on a small bug. 
My firmware definitely reset to 1.4.0 because the watch was off for a while. But in that case it shouldn't show me buttons to `validate` or `reset` since it's already reset. Maybe that was just a bug in the old 1.4.0 version that doesn't exist anymore.

Thanks again mate and thanks for all your work on WatchMate 🙏
Keep going, this is a simple but awesome application !
